pkg/operator/encryption/controllers/condition_controller.go

```diff
@@ -0,0 +1,242 @@
+
+	return c
+}
+
+func (c *conditionController) sync() error {
+	if ready, err := shouldRunEncryptionController(c.operatorClient); err != nil || !ready {
+		return err // we will get re-kicked when the operator status updates
+	}
+
+	currentConfig, desiredState, foundSecrets, transitioningReason, err := statemachine.GetEncryptionConfigAndState(c.deployer, c.secretClient, c.encryptionSecretSelector, c.encryptedGRs)
+	if err != nil || len(transitioningReason) > 0 {
+		return err
+	}
+
+	cond := operatorv1.OperatorCondition{
+		Type:    "Encrypted",
+		Status:  operatorv1.ConditionTrue,
+		Reason:  "EncryptionComplete",
+		Message: fmt.Sprintf("Resources all encrypted: %s", grString(c.encryptedGRs)),
+	}
+	currentState, _ := encryptionconfig.ToEncryptionState(currentConfig, foundSecrets)
+
+	if len(foundSecrets) == 0 {
+		cond.Status = operatorv1.ConditionFalse
+		cond.Reason = "EncryptionDisabled"
+		cond.Message = "Encryption is not enabled"
+	} else {
+		// check for identity key in desired state first. This will make us catch upcoming decryption early before
+		// it settles into the current config.
+		for _, s := range desiredState {
+			if s.WriteKey.Mode != state.Identity {
+				continue
+			}
+
+			if allMigrated(c.encryptedGRs, s.WriteKey.Migrated.Resources) {
+				cond.Status = operatorv1.ConditionFalse
+				cond.Reason = "EverythingDecrypted"
+				cond.Message = "Encryption mode set to identity and everything is decrypted"
+			} else {
+				cond.Status = operatorv1.ConditionFalse
+				cond.Reason = "EncryptionBeingDisabled"
+				cond.Message = "Encryption mode set to identity and decryption is not finished"
+			}
+			break
+		}
+	}
+	if cond.Status == operatorv1.ConditionTrue {
+		// now that the desired state look like it won't lead to identity as write-key, test the current state
+	NextResource:
+		for _, gr := range c.encryptedGRs {
+			s, ok := currentState[gr]
+			if !ok {
+				cond.Status = operatorv1.ConditionFalse
+				cond.Reason = "EncryptionIncomplete"
+				cond.Message = fmt.Sprintf("Resource %s is not encrypted", gr.String())
+				break NextResource
+			}
+
+			if s.WriteKey.Mode == state.Identity {
+				if allMigrated(c.encryptedGRs, s.WriteKey.Migrated.Resources) {
+					cond.Status = operatorv1.ConditionFalse
+					cond.Reason = "EverythingDecrypted"
+					cond.Message = "Encryption mode set to identity and everything is decrypted"
+				} else {
+					cond.Status = operatorv1.ConditionFalse
+					cond.Reason = "EncryptionBeingDisabled"
+					cond.Message = "Encryption mode set to identity and decryption is not finished"
+				}
+				break
+			}
+
+			// go through read keys until we find a completely migrated one. Finding an identity mode before
+			// means migration is ongoing. :
+			for _, rk := range s.ReadKeys {
+				if rk.Mode == state.Identity {
+					cond.Status = operatorv1.ConditionFalse
+					cond.Reason = "EncryptionBeingEnabled"
+					cond.Message = "Encryption is ongoing"
```

Comment:
this means that not all servers have observed the same `revision/encryptionconfig`

Reply:
no, it means we have an identity mode with a higher revision than the last migrated key. Can happen if keys are deleted.